src/interpreter/utils.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  type BlockStatement,
+  type ExternalDeclaration,
+  type FunctionDeclaration,
+  type Identifier,
+  type VariableDeclaration
+} from '../ast/types';
+import { type VirtualMemory } from '../memory/virtualMemory';
+import { type Closure, type Environment } from './types/interpreter';
+import { type Value } from './types/evaluationResults';
+import { isConstant, isVariableDeclaration } from '../ast/typeGuards';
+import { InvalidFunctionApplication } from './errors';
+
+const allocateUninitializedVariable = (memory: VirtualMemory): number => {
```

Comment:
Will we ever have a notion of variable size? Or is this just a simplification (all variables have the same size) to get things up and running first?

Reply:
Yes it's a simplification to get things up and running first. I think everything would be the same size/not complicated until we handle strings.

Reply:
> I think everything would be the same size/not complicated until we handle strings.

Hmm, but we potentially have to deal with types for the various sizes. For example, `uint8_t`, `uint16_t`, etc.